Source/missiles.cpp

```diff
@@ -2410,34 +2491,56 @@ void AddStone(MissileStruct &missile, Point dst, int /*midir*/)
 	UseMana(missile._misource, SPL_STONE);
 }
 
-void AddGolem(MissileStruct &missile, Point dst, int /*midir*/)
+void AddGolem(MissileStruct &missile, Point dst, Direction /*midir*/)
 {
+	missile._miDelFlag = true;
+
 	for (int i = 0; i < ActiveMissileCount; i++) {
 		int mx = ActiveMissiles[i];
 		if (Missiles[mx]._mitype == MIS_GOLEM) {
 			if ((&Missiles[mx] != &missile) && Missiles[mx]._misource == missile._misource) {
-				missile._miDelFlag = true;
 				return;
 			}
 		}
 	}
-	missile._miVar1 = missile.position.start.x;
-	missile._miVar2 = missile.position.start.y;
-	missile._miVar4 = dst.x;
-	missile._miVar5 = dst.y;
 	if (Monsters[missile._misource].position.tile != GolemHoldingCell && missile._misource == MyPlayerId)
 		M_StartKill(missile._misource, missile._misource);
+
 	UseMana(missile._misource, SPL_GOLEM);
+
+	int src = missile._misource;

```

Comment:
Might as well move this to the start of the function so it's used for the other `missile._misource` references. Could use a new name too (`playerId`?).